### ISSUE TITLE
Legacy OSDs on devices are being skipped in the upgrade to v1.2.3

### ISSUE THREAD

**Issue by travisn** [rook/rook#4797]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
The upgrade integration test is failing in the `release-1.2` branch since [build 54](https://jenkins.rook.io/blue/organizations/jenkins/rook%2Frook/detail/release-1.2/54/pipeline/).

It seems the upgrade is jumping straight from v1.0 to master, skipping over the v1.1 upgrade, which is needed to complete the smooth upgrade.

For example, this [test log](https://jenkins.rook.io/blue/rest/organizations/jenkins/pipelines/rook/pipelines/rook/branches/release-1.2/runs/61/nodes/45/steps/126/log/?start=0) shows that the operator image is set to `rook/ceph:master`. Then the test times out after waiting for the OSDs to start. The mons are actually able to upgrade, but the osds are not upgrade.
```
2020-01-31 22:55:52.201064 I | exec: Running command: kubectl -n upgrade-ns-system set image deploy/rook-ceph-operator rook-ceph-operator=rook/ceph:master
2020-01-31 22:56:07.312480 I | testutil: waiting for 3 deployments (found 0) with label app=rook-ceph-mon,rook-version!=v1.0.6 in namespace upgrade-ns
...
2020-01-31 22:56:42.353886 I | testutil: found 3 of 3 deployments with label app=rook-ceph-mon,rook-version!=v1.0.6 in namespace upgrade-ns
2020-01-31 22:56:42.357108 I | testutil: all 3 deployments with label app=rook-ceph-mon,rook-version!=v1.0.6 are running
2020-01-31 22:56:42.359428 I | testutil: waiting for 2 deployments (found 0) with label app=rook-ceph-osd,rook-version!=v1.0.6 in namespace upgrade-ns
2020-01-31 22:56:47.362283 I | testutil: waiting for 2 deployments (found 0) with label app=rook-ceph-osd,rook-version!=v1.0.6 in namespace upgrade-ns
2020-01-31 22:56:52.365145 I | testutil: waiting for 2 deployments (found 0) with label app=rook-ceph-osd,rook-version!=v1.0.6 in namespace upgrade-ns
...
```

**Expected behavior:**
Integration tests must pass

**How to reproduce it (minimal and precise):**

Run the Ceph upgrade integration test in the release-1.2 branch

**Comment by travisn** [comment]
The root cause of the failing test is that the legacy OSDs that were created on Rook partitions are not getting upgraded. With the new partitioning support, the devices with the Rook partitions are being skipped. In the upgrade test in the release-1.2 branch, the upgrade test configures OSDs with v13.2.0, which doesn't have c-v support yet. Thus, when upgrading to v1.2 the partitioning detection causes these to be skipped. We still need support for the legacy OSD upgrade until v1.3.

Here is the osd prepare log:
```
2020-02-01 09:01:38.779040 I | inventory: skipping device "sdc" because it has child, considering the child instead.
2020-02-01 09:01:38.779233 I | exec: Running command: lsblk /dev/sdc2 --bytes --nodeps --pairs --output SIZE,ROTA,RO,TYPE,PKNAME
2020-02-01 09:01:38.784014 I | exec: Running command: udevadm info --query=property /dev/sdc2
2020-02-01 09:01:38.785935 I | exec: Running command: lsblk /dev/sdc3 --bytes --nodeps --pairs --output SIZE,ROTA,RO,TYPE,PKNAME
2020-02-01 09:01:38.787755 I | exec: Running command: udevadm info --query=property /dev/sdc3
2020-02-01 09:01:38.789345 I | exec: Running command: lsblk /dev/sdc1 --bytes --nodeps --pairs --output SIZE,ROTA,RO,TYPE,PKNAME
2020-02-01 09:01:38.791416 I | exec: Running command: udevadm info --query=property /dev/sdc1
2020-02-01 09:01:38.793272 I | exec: Running command: lsblk /dev/sda --bytes --nodeps --pairs --output SIZE,ROTA,RO,TYPE,PKNAME
2020-02-01 09:01:38.795763 I | exec: Running command: sgdisk --print /dev/sda
2020-02-01 09:01:38.797927 I | exec: Running command: udevadm info --query=property /dev/sda
2020-02-01 09:01:38.799575 I | exec: Running command: lsblk --noheadings --pairs /dev/sda
2020-02-01 09:01:38.803113 I | inventory: skipping device "sda" because it has child, considering the child instead.
2020-02-01 09:01:38.803127 I | exec: Running command: lsblk /dev/sda1 --bytes --nodeps --pairs --output SIZE,ROTA,RO,TYPE,PKNAME
2020-02-01 09:01:38.805039 I | exec: Running command: udevadm info --query=property /dev/sda1
2020-02-01 09:01:38.806604 I | exec: Running command: lsblk /dev/loop7 --bytes --nodeps --pairs --output SIZE,ROTA,RO,TYPE,PKNAME
2020-02-01 09:01:38.808286 W | inventory: skipping device loop7: Failed to complete 'lsblk /dev/loop7': exit status 1. 
2020-02-01 09:01:38.808297 I | exec: Running command: lsblk /dev/loop5 --bytes --nodeps --pairs --output SIZE,ROTA,RO,TYPE,PKNAME
2020-02-01 09:01:38.810016 W | inventory: skipping device loop5: Failed to complete 'lsblk /dev/loop5': exit status 1. 
2020-02-01 09:01:38.810030 I | exec: Running command: lsblk /dev/loop3 --bytes --nodeps --pairs --output SIZE,ROTA,RO,TYPE,PKNAME
2020-02-01 09:01:38.811684 W | inventory: skipping device loop3: Failed to complete 'lsblk /dev/loop3': exit status 1. 
2020-02-01 09:01:38.811694 I | cephosd: creating and starting the osds
2020-02-01 09:01:38.811700 I | cephosd: skipping device "sdb2" because it is a partition and ceph version is too old, you need at least ceph "14.2.8-0 nautilus"
2020-02-01 09:01:38.811703 I | cephosd: skipping device "sdb3" because it is a partition and ceph version is too old, you need at least ceph "14.2.8-0 nautilus"
2020-02-01 09:01:38.811707 I | cephosd: skipping device "sdb1" because it is a partition and ceph version is too old, you need at least ceph "14.2.8-0 nautilus"
2020-02-01 09:01:38.811709 I | cephosd: skipping device "sdc2" because it is a partition and ceph version is too old, you need at least ceph "14.2.8-0 nautilus"
2020-02-01 09:01:38.811712 I | cephosd: skipping device "sdc3" because it is a partition and ceph version is too old, you need at least ceph "14.2.8-0 nautilus"
2020-02-01 09:01:38.811714 I | cephosd: skipping device "sdc1" because it is a partition and ceph version is too old, you need at least ceph "14.2.8-0 nautilus"
2020-02-01 09:01:38.811717 I | cephosd: skipping device "sda1" because it is a partition and ceph version is too old, you need at least ceph "14.2.8-0 nautilus"
2020-02-01 09:01:38.816862 I | cephosd: configuring osd devices: {"Entries":{}}
```

**Comment by travisn** [comment]
Fixed by #4799
